File: Objective-C/0002.m
#import <Foundation/Foundation.h>

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        // Define the maximum value
        const int maxVal = 4000000;
        
        // Initialize the first two terms of the Fibonacci sequence
        int term1 = 1;
        int term2 = 2;
        int sumEven = 0;
        
        // Compute the sum of even Fibonacci numbers
        while (term1 <= maxVal) {
            // Check if the current term is even
            if (term1 % 2 == 0) {
                sumEven += term1;
            }
            
            // Generate the next term in the Fibonacci sequence
            int nextTerm = term1 + term2;
            term1 = term2;
            term2 = nextTerm;
        }
        
        // Output the result
        NSLog(@"Sum of the even-valued terms: %d", sumEven);
    }
    return 0;
}

File: Haskell/0002.hs
-- Function to generate Fibonacci sequence
fibonacci :: [Integer]
fibonacci = 1 : 2 : zipWith (+) fibonacci (tail fibonacci)

-- Function to sum even Fibonacci numbers up to a given maximum value
sumEvenFibonacci :: Integer -> Integer
sumEvenFibonacci maxVal = sum [x | x <- takeWhile (<= maxVal) fibonacci, even x]

-- Main function to compute and print the result
main :: IO ()
main = do
    let maxValue = 4000000
    let result = sumEvenFibonacci maxValue
    print result
